contrib/mesos/pkg/node/node.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	log "github.com/golang/glog"
+	mesos "github.com/mesos/mesos-go/mesosproto"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	LabelPrefix = "k8s.mesosphere.io/"
```

Comment:
at some point it might be nice to centralize our label munging so that it's
not scattered across our codebase; in a follow-up PR

On Mon, Sep 14, 2015 at 7:41 AM, James DeFelice <james@mesosphere.io> wrote:

> i'd rather not prefix the prefix with `attributes.` since that creates a
> new vendor namespace; we should stick with `k8s.mesosphere.io` since we
> use that namespace elsewhere (and this has specific meaning in k8s -- this
> is essentially our vendor tag). the `*` is fine as a regexp filter but not
> when you use StartsWith for testing and string concatenation for building
> new label names. and i'd rather not use regexp filter, it's overkill for
> what we need. strongly urge something along the lines of the format that I
> suggested.
>
> On Mon, Sep 14, 2015 at 7:35 AM, Dr. Stefan Schimanski <
> notifications@github.com> wrote:
>
>> In contrib/mesos/pkg/node/node.go
>> <https://github.com/kubernetes/kubernetes/pull/13857#discussion_r39383946>
>> :
>>
>> > +
>> > +import (
>> > +	"encoding/json"
>> > +	"fmt"
>> > +	"strconv"
>> > +	"strings"
>> > +
>> > +	log "github.com/golang/glog"
>> > +	mesos "github.com/mesos/mesos-go/mesosproto"
>> > +	"k8s.io/kubernetes/pkg/api"
>> > +	"k8s.io/kubernetes/pkg/api/errors"
>> > +	client "k8s.io/kubernetes/pkg/client/unversioned"
>> > +)
>> > +
>> > +const (
>> > +	LabelPrefix = "k8s.mesosphere.io/"
>>
>> attributes.k8s.mesosphere.io/* ?
>>
>> I don't like to put something behind the /
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/kubernetes/kubernetes/pull/13857/files#r39383946>.
>>
>
>